lib/internal/streams/transform.js

```diff
@@ -244,3 +246,22 @@ Transform.prototype._read = function() {
     callback();
   }
 };
+
+Transform.by = function by(asyncGeneratorFn, options) {
+  let { promise, resolve } = createDeferredPromise();
+  return from(Duplex, asyncGeneratorFn(async function*() {
+    while (true) {
+      const { chunk, done, cb } = await promise;
+      process.nextTick(cb);
+      if (done) return;
+      yield chunk;
+      ({ promise, resolve } = createDeferredPromise());
+    }
+  }()), {
+    objectMode: true,
+    autoDestroy: true,
+    ...options,
+    write: (chunk, encoding, cb) => resolve({ chunk, done: false, cb }),
```

Comment:
```suggestion
    write: (chunk, encoding, cb) => resolve({ chunk, encoding, done: false, cb }),
```

how about this?

Reply:
this would just make encoding available here:
```js
const { chunk, done, cb } = await promise;
```